cmd/observer/node_builder/observer_builder.go

```diff
@@ -991,10 +992,15 @@ func (builder *ObserverServiceBuilder) enqueueRPCServer() {
 		}
 
 		proxy := &apiproxy.FlowAccessAPIRouter{
-			Logger:          builder.Logger,
-			Metrics:         metrics.NewObserverCollector(),
-			Upstream:        forwarder,
-			AccessAPIServer: engineBuilder.Handler(),
+			Logger:   builder.Logger,
+			Metrics:  metrics.NewObserverCollector(),
+			Upstream: forwarder,
+			Observer: protocol.NewHandler(protocol.New(
+				node.State,
+				node.Storage.Blocks,
+				node.Storage.Headers,
+				backend.NewNetworkAPI(node.State, node.RootChainID, backend.DefaultSnapshotHistoryLimit),
```

Comment:
Why did we replace AccessAPIServer with o\Observer?

Reply:
The AccessAPIServer was what the Observer used to handle requests without using the upstream access node. But the Observer should not support the full access api.